stream_alert/shared/lookup_tables/driver_s3.py

```diff
@@ -0,0 +1,199 @@
+            )
+            return
+
+        LOGGER.info(
+            'LookupTable (%s): Needs refresh, starting now. Last refresh: %s; Currently: %s',
+            self.id,
+            self._last_load_time,
+            now
+        )
+        self._reload()
+
+    def _reload(self):
+        """
+        Reaches to AWS S3, downloads the relevant data file, decompresses, decodes, and caches
+        the file's contents into memory.
+        """
+        # First, download the item from S3
+        try:
+            start_time = time.time()
+            s3_object = self._s3_client.Object(self._s3_bucket, self._s3_key).get()
+            data = s3_object.get('Body').read()
+
+            total_time = time.time() - start_time
+            size_kb = round(s3_object.get('ContentLength') / 1024.0, 2)
+            size_mb = round(size_kb / 1024.0, 2)
+            LOGGER.debug(
+                'LookupTable (%s): Downloaded S3 file size %s in %s seconds',
+                self.id,
+                '{}MB'.format(size_mb) if size_mb else '{}KB'.format(size_kb),
+                round(total_time, 2)
+            )
+        except ClientError as err:
+            LOGGER.error(
+                'LookupTable (%s): Encountered error while downloading %s from %s: %s',
+                self.id,
+                self._s3_key,
+                self._s3_bucket,
+                err.response['Error']['Message']
+            )
+            raise LookupTablesInitializationError(
+                'LookupTable S3 Driver Failed with Message: {}'.format(
+                    err.response['Error']['Message']
+                )
+            )
+
+        except (ConnectTimeoutError, ReadTimeoutError):
+            # Catching ConnectTimeoutError and ReadTimeoutError from botocore
+            LOGGER.error(
+                'LookupTable (%s): Reading from S3 timed out',
+                self.id
+            )
+            raise LookupTablesInitializationError(
+                'LookupTable ({}): Reading from S3 timed out'.format(self.id)
+            )
+
+        # The lookup data can optionally be compressed, so try to decompress
+        # This will fall back and use the original data if decompression fails
+        if self._compression:
+            try:
+                data = zlib.decompress(data, 47)
+                LOGGER.debug(
+                    'LookupTable (%s): Object decompressed to %d byte payload',
+                    self.id,
+                    sys.getsizeof(data)
+                )
+            except zlib.error:
+                LOGGER.warn(
+                    'LookupTable (%s): Data is not compressed; defaulting to original payload',
+                    self.id
+                )
+        else:
+            LOGGER.debug('LookupTable (%s): File does not need decompression')
+
+        # Decode the data; right now we make the assumption that the data is always encoded
+        # as JSON.
+        try:
+            self._s3_data = json.loads(data)
```

Comment:
should we support other "formats" as well? like yaml or ini? idk if we should

Reply:
Definitely could support them!